community_tasks/jglue_evals.py

```diff
@@ -0,0 +1,341 @@
+# SOFTWARE.
+"""
+The Japanese benchmark JGLUE has been implemented, covering four of its five
+benchmark tasks (MARC-ja is unavailable as the dataset has been removed at
+Amazon's request). This effort is part of the plan to reimplement llm-jp-eval
+in the lighteval framework.
+
+Tasks:
+- JSTS
+- JNLI
+- JSQUAD
+- JCommonsenseQA
+
+The datasets were developed by Yahoo Japan, with prompts inspired by
+Stability-AI's fork of lm-evaluation-harness from last year, though the fork is
+no longer directly runnable.
+
+Dataset: https://github.com/yahoojapan/JGLUE
+Prompts: https://github.com/Stability-AI/lm-evaluation-harness
+"""
+
+import numpy as np
+from scipy.stats import pearsonr, spearmanr
+
+from lighteval.metrics.metrics import CorpusLevelMetric, Metrics
+from lighteval.metrics.utils.metric_utils import MetricCategory, MetricUseCase
+from lighteval.tasks.lighteval_task import LightevalTaskConfig
+from lighteval.tasks.requests import Doc
+
+
+_CITATION = """
+@inproceedings{kurihara-etal-2022-jglue,
+    title = "{JGLUE}: {J}apanese General Language Understanding Evaluation",
+    author = "Kurihara, Kentaro  and
+      Kawahara, Daisuke  and
+      Shibata, Tomohide",
+    booktitle = "Proceedings of the Thirteenth Language Resources and Evaluation Conference",
+    month = jun,
+    year = "2022",
+    address = "Marseille, France",
+    publisher = "European Language Resources Association",
+    url = "https://aclanthology.org/2022.lrec-1.317",
+    pages = "2957--2966",
+    abstract = "To develop high-performance natural language understanding (NLU) models, it is necessary to have a benchmark to evaluate and analyze NLU ability from various perspectives. While the English NLU benchmark, GLUE, has been the forerunner, benchmarks are now being released for languages other than English, such as CLUE for Chinese and FLUE for French; but there is no such benchmark for Japanese. We build a Japanese NLU benchmark, JGLUE, from scratch without translation to measure the general NLU ability in Japanese. We hope that JGLUE will facilitate NLU research in Japanese.",
+}
+"""
+
+
+# Metrics
+def correlation_metric(golds: list[int], predictions: list[str], **kwargs):
+    def convert_to_float(score):
+        try:
+            return float(score)
+        except ValueError:
+            return None
+
+    predicted_score = convert_to_float(predictions[0])
+    gold_score = convert_to_float(golds[0])
+
+    return {
+        "predicted_score": predicted_score,
+        "gold_score": gold_score,
+    }
+
+
+def spearman_corpus_metric(items):
+    predicted_scores, gold_scores = zip(
+        *[
+            (item["predicted_score"], item["gold_score"])
+            for item in items
+            if (item["gold_score"] is not None and item["predicted_score"] is not None)
+        ]
+    )
+    r, _ = spearmanr(predicted_scores, gold_scores)
+    if np.isnan(r):
+        return 0.0
+    frac = len(predicted_scores) / len(items)
```

Comment:
@NathanHB I believe we could add these 2 to core metrics, wdyt?

Reply:
definitely !